apps/src/lib/node/ledger/mod.rs

```diff
@@ -310,43 +310,79 @@ async fn run_aux(config: config::Ledger, wasm_dir: PathBuf) {
     let (broadcaster_sender, broadcaster_receiver) =
         tokio::sync::mpsc::unbounded_channel();
 
-    let ethereum_url = config.ethereum_url();
     let (ethereum_node, oracle, broadcaster) = if matches!(
         config.tendermint.tendermint_mode,
         TendermintMode::Validator
     ) {
-        // boot up the ethereum node process and wait for it to finish syncing
-        let (eth_sender, eth_receiver) = unbounded_channel();
-        let url = ethereum_url.clone();
-        let (ethereum_node, abort_sender) = EthereumNode::new(&url)
-            .await
-            .expect("Unable to start the Ethereum fullnode");
-
-        // Start Ethereum fullnode
-        // Channel for signalling shut down to Tendermint process
+        let abort_send_for_eth = abort_send.clone();
         let (eth_abort_send, eth_abort_recv) =
             tokio::sync::oneshot::channel::<tokio::sync::oneshot::Sender<()>>();
-        let abort_send_for_eth = abort_send.clone();
-        // run geth in the background
-        let ethereum_node = tokio::spawn(async move {
-            // On panic or exit, the `Drop` of `AbortSender` will send abort
-            // message
-            let aborter = Aborter {
-                sender: abort_send_for_eth,
-                who: "Ethereum",
-            };
-
-            let res = ethereum_node::run(ethereum_node, eth_abort_recv)
-                .map_err(Error::Ethereum)
-                .await;
-            tracing::info!("Ethereum fullnode is no longer running.");
-
-            drop(aborter);
-            res
-        });
-
-        let oracle =
-            ethereum_node::run_oracle(ethereum_url, eth_sender, abort_sender);
+        // we want any Ethereum endpoint set in the environment to override the
+        // one set in the config
+        let ethereum_url = std::env::var(ENV_VAR_ETHEREUM_URL).ok();
+        let (ethereum_url, ethereum_node, abort_sender) = match ethereum_url {
+            Some(ethereum_url) => {
+                tracing::info!(
+                    %ethereum_url,
+                    "Ethereum URL provided in environment, will not start geth"
+                );
+                // we return a dummy join handle and sender, so we can share
+                // logic further down this function with how we'd normally
+                // handle a `geth` subprocess
+                let handle = tokio::spawn(async {
+                    tracing::info!(
+                        "Spawning a task to await abort signal for Ethereum \
+                         node (although we are not running one)"
+                    );
+                    eth_abort_recv.await.unwrap();
+                    tracing::info!(
+                        "Received abort signal that was intended for Ethereum \
+                         node"
+                    );
```

Comment:
This is not great, the reason I'm spawning a task which is essentially just a noop here is because I didn't want to change code or refactor further down the function (especially since there is another refactor in flight for `run_aux`), also I need to learn more about `tokio`.